bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/openhab/core/io/rest/sitemap/internal/SitemapResource.java

```diff
@@ -341,26 +405,31 @@ public Object createEventSubscription() {
     @Produces(MediaType.SERVER_SENT_EVENTS)
     @Operation(operationId = "getSitemapEvents", summary = "Get sitemap events.", responses = {
             @ApiResponse(responseCode = "200", description = "OK"),
-            @ApiResponse(responseCode = "400", description = "Page not linked to the subscription."),
+            @ApiResponse(responseCode = "400", description = "Missing sitemap or page parameter, or page not linked successfully to the subscription."),
             @ApiResponse(responseCode = "404", description = "Subscription not found.") })
     public void getSitemapEvents(@Context final SseEventSink sseEventSink, @Context final HttpServletResponse response,
             @PathParam("subscriptionid") @Parameter(description = "subscription id") String subscriptionId,
             @QueryParam("sitemap") @Parameter(description = "sitemap name") @Nullable String sitemapname,
             @QueryParam("pageid") @Parameter(description = "page id") @Nullable String pageId) {
+        getSitemapEvents(sseEventSink, response, subscriptionId, sitemapname, pageId, false);
         logger.debug("Received HTTP GET request from IP {} at '{}' for sitemap {} and page {}", request.getRemoteAddr(),
                 uriInfo.getPath(), sitemapname, pageId);
+    }
 
+    private void getSitemapEvents(SseEventSink sseEventSink, HttpServletResponse response, String subscriptionId,
+            @Nullable String sitemapname, @Nullable String pageId, boolean subscribeToWholeSitemap) {
         final SseSinkInfo sinkInfo = knownSubscriptions.get(subscriptionId);
         if (sinkInfo == null) {
             logger.debug("Subscription id {} does not exist.", subscriptionId);
             response.setStatus(Status.NOT_FOUND.getStatusCode());
             return;
         }
-        if (sitemapname != null && pageId != null) {
-            subscriptions.setPageId(subscriptionId, sitemapname, pageId);
+        if (sitemapname != null && (subscribeToWholeSitemap || pageId != null)) {
+            subscriptions.updateSubscriptionLocation(subscriptionId, sitemapname, pageId);
         }
-        if (subscriptions.getSitemapName(subscriptionId) == null || subscriptions.getPageId(subscriptionId) == null) {
-            logger.debug("Subscription id {} is not yet linked to a sitemap/page.", subscriptionId);
+        if (subscriptions.getSitemapName(subscriptionId) == null
+                || subscriptions.getPageId(subscriptionId) == null && !subscribeToWholeSitemap) {
```

Comment:
I believe some parenthesis are missing around the second line. Please confirm.

Reply:
good catch

Reply:
No, because && has higher operator precedence.

Reply:
The condition says "we need to have successfully subscribed to a sitemap and, if we do not want to subscribe to a whole sitemap, we need to have successfully subscribed to a page"

Reply:
Even if the code is good due to precedence of such operator over another, I will prefer you add parenthesis as precedence between && and || is not something very intuitive.

Reply:
Matter of taste, I always prefer less boilerplate but will follow your preference here as you´re the maintainers :) To explain my preference, I think in complicated boolean statements, too many parenthesis in a row confuse new developers, as they have to count the parenthesis to know what they enclose. I think, although not coming from a Python background, that smart indentations help understanding much better. For that statement for example I would prefer the following, but I believe spotless prohibits that:

```Java
if (subscriptions.getSitemapName(subscriptionId) == null
     || subscriptions.getPageId(subscriptionId) == null 
        && !subscribeToWholeSitemap)
```